src/main/java/org/kohsuke/stapler/idea/extension/JavaStructureViewExtension.java

```diff
@@ -0,0 +1,50 @@
+package org.kohsuke.stapler.idea.extension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.intellij.ide.structureView.StructureViewTreeElement;
+import com.intellij.ide.structureView.impl.xml.XmlFileTreeElement;
+import com.intellij.openapi.roots.ProjectFileIndex;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiJavaFile;
+import com.intellij.psi.PsiManager;
+import com.intellij.psi.xml.XmlFile;
+import org.kohsuke.stapler.idea.psi.JellyFile;
+
+/**
+ * Additional jelly files in java structure view
+ * Items are retrieved in related src/main/java directory
+ *
+ * @author Julien Greffe
+ */
+public class JavaStructureViewExtension extends AbstractStructureViewExtension {
+
+    @Override
+    public Class<? extends PsiElement> getType() {
+        return PsiJavaFile.class;
```

Comment:
Views are stabled to the class, so I do not think this is accurate. I think this should be `com.intellij.psi.PsiClass`.

Reply:
Correct, will fix.